cmd/network-tracer/main.go

```diff
@@ -0,0 +1,153 @@
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/pidfile"
+	"github.com/DataDog/datadog-process-agent/config"
+)
+
+// Flag values
+var opts struct {
+	configPath string
+
+	pidFilePath string
+	debug       bool
+	version     bool
+}
+
+// Version info sourced from build flags
+var (
+	GoVersion string
+	Version   string
+	GitCommit string
+	GitBranch string
+	BuildDate string
+)
+
+func main() {
+	// Parse flags
+	flag.StringVar(&opts.configPath, "config", "/etc/datadog-agent/network-tracer.yaml", "Path to network-tracer config formatted as YAML")
+	flag.StringVar(&opts.pidFilePath, "pid", "", "Path to set pidfile for process")
+	flag.BoolVar(&opts.version, "version", false, "Print the version and exit")
+	flag.Parse()
+
+	// Set up a default config before parsing config so we log errors nicely.
+	// The default will be stdout since we can't assume any file is writeable.
+	if err := config.NewLoggerLevel("info", "", true); err != nil {
+		panic(err)
+	}
+	defer log.Flush()
+
+	// --version
+	if opts.version {
+		fmt.Println(versionString())
+		os.Exit(0)
+	}
+
+	// --pid
+	if opts.pidFilePath != "" {
+		if err := pidfile.WritePID(opts.pidFilePath); err != nil {
+			log.Errorf("Error while writing PID file, exiting: %v", err)
+			os.Exit(1)
+		}
+
+		log.Infof("pid '%d' written to pid file '%s'", os.Getpid(), opts.pidFilePath)
+
+		defer func() {
+			os.Remove(opts.pidFilePath)
+		}()
+	}
+
+	// Parsing INI and/or YAML config files
+	cfg := parseConfig()
+
+	// Exit if network tracer is disabled
+	if !cfg.EnableNetworkTracing {
+		log.Info("network tracer not enabled. exiting.")
+		gracefulExit()
+	}
+
+	nettracer, err := CreateNetworkTracer(cfg)
+	if err != nil && strings.HasPrefix(err.Error(), ErrTracerUnsupported.Error()) {
+		// If tracer is unsupported by this operating system, then exit gracefully
+		log.Infof("%s, exiting.", err)
+		gracefulExit()
+	} else if err != nil {
+		log.Criticalf("failed to create network tracer: %s", err)
+		os.Exit(1)
+	}
+	defer nettracer.Close()
+
+	go nettracer.Run()
+	log.Infof("network tracer started")
+
+	// Handles signals, which tells us whether we should exit.
+	e := make(chan bool)
+	go handleSignals(e)
+	<-e
+}
+
+func gracefulExit() {
+	// A sleep is necessary to ensure that supervisor registers this process as "STARTED"
+	// If the exit is "too quick", we enter a BACKOFF->FATAL loop even though this is an expected exit
+	// http://supervisord.org/subprocess.html#process-states
+	time.Sleep(5 * time.Second)
+	os.Exit(0)
+}
+
+func handleSignals(exit chan bool) {
+	sigIn := make(chan os.Signal, 100)
+	signal.Notify(sigIn)
+	// unix only in all likelihood;  but we don't care.
+	for sig := range sigIn {
+		switch sig {
+		case syscall.SIGINT, syscall.SIGTERM, syscall.SIGKILL, syscall.SIGQUIT:
+			log.Criticalf("Caught signal '%s'; terminating.", sig)
+			close(exit)
+		default:
```

Comment:
I think it would be nice to have the same behavior as the process agent by logging even in the default case (for debugging purposes): https://github.com/DataDog/datadog-process-agent/blob/master/cmd/agent/main_nodocker.go#L24